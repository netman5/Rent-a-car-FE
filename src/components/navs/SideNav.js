import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  FaTwitter, FaFacebookF, FaPinterestP, FaVimeoV,
} from 'react-icons/fa';
import { TiSocialGooglePlus } from 'react-icons/ti';
import styles from './SideNav.module.css';

const adminLinks = [
  { id: 1, path: '/', text: 'MODELS' },
  { id: 2, path: '/reserve', text: 'RESERVE' },
  { id: 3, path: '/reservation', text: 'MY RESERVATION' },
  { id: 4, path: '/add-car', text: 'ADD CAR' },
  { id: 5, path: '/delete', text: 'DELETE' },
];

const links = [
  { id: 1, path: '/', text: 'MODELS' },
  { id: 2, path: '/reserve', text: 'RESERVE' },
  { id: 3, path: '/reservation', text: 'MY RESERVATION' },
];

const SocialMedia = () => (
  <div className="position-absolute bottom-0 start-0">
    <div className={styles.socialmedia}>
      <FaTwitter className={styles.icons} />
      <FaFacebookF className={styles.icons} />
      <TiSocialGooglePlus className={styles.icons} />
      <FaVimeoV className={styles.icons} />
      <FaPinterestP className={styles.icons} />
    </div>
    <div>
      <p className={styles.footer}>&copy; 2020 Rent A Car Company</p>
    </div>
  </div>
);

const SideNav = () => {
  const data = JSON.parse(localStorage.getItem('user'));
  const { user } = data || {};
  return (
    <div className="bg-light">
      <div className={styles.sidebar}>
        <ul className={styles.navlinks}>
          {user && user.role === 'admin' ? (adminLinks.map((link) => (
            <li key={link.id}>
              <NavLink
                to={link.path}
                className="btn btn-light"
                style={({ isActive }) => (isActive
                  ? {
                    color: '#fff',
                    backgroundColor: '#97bf0e',
                    width: '100%',
                  }
                  : { color: '#3a3a3a', width: '100%' })}
              >
                {link.text}
              </NavLink>
            </li>
          ))) : links.map((link) => (
            <li key={link.id}>
              <NavLink
                to={link.path}
                className="btn btn-light"
                style={({ isActive }) => (isActive
                  ? {
                    color: '#fff',
                    backgroundColor: '#97bf0e',
                    width: '100%',
                  }
                  : { color: '#3a3a3a', width: '100%' })}
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
        <SocialMedia />
      </div>
    </div>
  );
};

export default SideNav;
