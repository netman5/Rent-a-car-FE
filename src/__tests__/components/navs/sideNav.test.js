import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import SideNav from '../../../components/navs/SideNav';
import store from '../../../redux/configureStore';

it('Cars renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <SideNav />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
