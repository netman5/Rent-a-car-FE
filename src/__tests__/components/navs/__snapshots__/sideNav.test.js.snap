// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cars renders correctly 1`] = `
<div
  className="bg-light"
>
  <div
    className="sidebar"
  >
    <ul
      className="navlinks"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#97bf0e",
              "color": "#fff",
            }
          }
        >
          MODELS
        </a>
      </li>
      <li>
        <a
          className=""
          href="/reserve"
          onClick={[Function]}
          style={
            Object {
              "color": "#3a3a3a",
            }
          }
        >
          RESERVE
        </a>
      </li>
      <li>
        <a
          className=""
          href="/reservation"
          onClick={[Function]}
          style={
            Object {
              "color": "#3a3a3a",
            }
          }
        >
          MY RESERVATION
        </a>
      </li>
    </ul>
    <div
      className="position-absolute bottom-0 start-0"
    >
      <div
        className="socialmedia"
      >
        <svg
          className="icons"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          />
        </svg>
        <svg
          className="icons"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
          />
        </svg>
        <svg
          className="icons"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.9 13.5l-.7-.5c-.2-.2-.5-.4-.5-.8s.3-.7.6-1c.8-.6 1.7-1.3 1.7-2.8 0-1.5-.9-2.3-1.4-2.7h1.2l1.2-.7h-4.1c-1 0-2.4.2-3.5 1.1-.8.7-1.2 1.7-1.2 2.6 0 1.5 1.2 3.1 3.3 3.1h.6c-.1.2-.2.4-.2.7 0 .6.3 1 .6 1.3-.9.1-2.5.2-3.8.9-1.2.7-1.5 1.7-1.5 2.4 0 1.5 1.4 2.8 4.2 2.8 3.4 0 5.2-1.9 5.2-3.7 0-1.3-.8-1.9-1.7-2.7zm-2.5-2.2c-1.7 0-2.5-2.2-2.5-3.5 0-.5.1-1 .4-1.5.3-.4.9-.7 1.4-.7 1.6 0 2.5 2.2 2.5 3.6 0 .4 0 1-.5 1.4-.3.4-.9.7-1.3.7zm0 7.9c-2.1 0-3.5-1-3.5-2.4s1.3-1.9 1.7-2c.8-.3 1.9-.3 2.1-.3h.5c1.5 1.1 2.1 1.6 2.1 2.6 0 1.2-1 2.1-2.9 2.1zM17 12h-2v-1h2v-1.9l1-.1v2h2v1h-2v2h-1z"
          />
        </svg>
        <svg
          className="icons"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M447.8 153.6c-2 43.6-32.4 103.3-91.4 179.1-60.9 79.2-112.4 118.8-154.6 118.8-26.1 0-48.2-24.1-66.3-72.3C100.3 250 85.3 174.3 56.2 174.3c-3.4 0-15.1 7.1-35.2 21.1L0 168.2c51.6-45.3 100.9-95.7 131.8-98.5 34.9-3.4 56.3 20.5 64.4 71.5 28.7 181.5 41.4 208.9 93.6 126.7 18.7-29.6 28.8-52.1 30.2-67.6 4.8-45.9-35.8-42.8-63.3-31 22-72.1 64.1-107.1 126.2-105.1 45.8 1.2 67.5 31.1 64.9 89.4z"
          />
        </svg>
        <svg
          className="icons"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M204 6.5C101.4 6.5 0 74.9 0 185.6 0 256 39.6 296 63.6 296c9.9 0 15.6-27.6 15.6-35.4 0-9.3-23.7-29.1-23.7-67.8 0-80.4 61.2-137.4 140.4-137.4 68.1 0 118.5 38.7 118.5 109.8 0 53.1-21.3 152.7-90.3 152.7-24.9 0-46.2-18-46.2-43.8 0-37.8 26.4-74.4 26.4-113.4 0-66.2-93.9-54.2-93.9 25.8 0 16.8 2.1 35.4 9.6 50.7-13.8 59.4-42 147.9-42 209.1 0 18.9 2.7 37.5 4.5 56.4 3.4 3.8 1.7 3.4 6.9 1.5 50.4-69 48.6-82.5 71.4-172.8 12.3 23.4 44.1 36 69.3 36 106.2 0 153.9-103.5 153.9-196.8C384 71.3 298.2 6.5 204 6.5z"
          />
        </svg>
      </div>
      <div>
        <p
          className="footer"
        >
          Â© 2020 Rent A Car Company
        </p>
      </div>
    </div>
  </div>
</div>
`;
